core/dbt/contracts/graph/nodes.py

```diff
@@ -365,6 +367,28 @@ def patch(self, patch: "ParsedNodePatch"):
         self.created_at = time.time()
         self.description = patch.description
         self.columns = patch.columns
+        # This might not be the ideal place to validate the "access" field,
+        # but at this point we have the information we need to properly
+        # validate and we don't before this.
+        if patch.access:
+            if self.resource_type == NodeType.Model:
+                if AccessType.is_valid(patch.access):
+                    self.access = AccessType(patch.access)
+                else:
+                    warn_or_error(
+                        InvalidValueForField(
+                            field_value=patch.access,
+                            field_name="access",
+                        )
+                    )
```

Comment:
Instead of a warning (which a user can promote to an error), Is it possible to raise a `ParsingError` here instead? I think that's more consistent with existing product expectations for other entities. For example, configuring `type: invalid` on an exposure gets: 

> dbt.exceptions.YamlParseDictError: Parsing Error
  Invalid exposures config given in FilePath(searched_path='models', relative_path='schema.yml', modification_time=1677270162.964944, project_root='/Users/michelleark/src/jaffle_shop') @ exposures: {'name': 'jaffle_shop_report', 'type': 'invalid', 'owner': {'name': 'testsss'}} - at path ['type']: 'invalid' is not one of ['dashboard', 'notebook', 'analysis', 'ml', 'application']

Reply:
Also, can we define `access` on the dataclass declaration for the `ParsedNode`, simialr to the other class attributes set by this method (created_at, description, patch_path). Ideally `access` is a `AccessType` if possible.

Reply:
Sure. It's an exception, so it aborts everything, which is not my favorite pattern because if you have multiple errors you have to address them one at a time, but we can certainly do that.

Reply:
I'm not sure what you're suggesting with regard to ParsedNode. It's set in the dataclass declaration for ModelNode, which is where you wanted it.

Reply:
>I'm not sure what you're suggesting with regard to ParsedNode. It's set in the dataclass declaration for ModelNode, which is where you wanted it.

I see, makes sense. Still getting my head around patches. Sorry for the confusion!

Reply:
> which is not my favorite pattern because if you have multiple errors you have to address them one at a time

Agree - but I'm nervous about gracefully handling this invalid config because, given that the default `protected` value is not the 'most' restricted access option. For example, a typo to `access:priivate` would lead to `access:protected` on the model, which is probably a more open level of access than the user meant to specify. Hopefully they'd catch it in review / via logs but I think it's worth raising an exception for this case.